### PR TITLE
[NEW] URL mappings for spring

### PR DESCRIPTION
**Pull request by monu-k2io**
Feature for getting url mappings for spring-webmvc and spring-boot

### CODE REVIEW

File: instrumentation-security/spring-webmvc-3.1.0/src/main/java/org/springframework/web/servlet/handler310/SpringHelper.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.util.Iterator;
+
+public class SpringHelper {
+    public static <T> void gatherURLMappings(T mapping){
+        try {
+            RequestMappingInfo mappingInfo = (RequestMappingInfo) mapping;
+            for (RequestMethod method : mappingInfo.getMethodsCondition().getMethods()) {
+                PatternsRequestCondition patternsCondition = mappingInfo.getPatternsCondition();
+                if (patternsCondition != null)
```
Comment (lovesh-ap) [inline instrumentation-security/spring-webmvc-3.1.0/src/main/java/org/springframework/web/servlet/handler310/SpringHelper.java:18]: This pattern check should be even before L:16 and please always add curly braces after if and for even if the code is one liner.

File: instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.util.Iterator;
+
+public class SpringHelper {
+    public static <T> void gatherURLMappings(T mapping){
+        try {
+            RequestMappingInfo mappingInfo = (RequestMappingInfo) mapping;
+            for (RequestMethod method : mappingInfo.getMethodsCondition().getMethods()) {
+                PatternsRequestCondition patternsCondition = mappingInfo.getPatternsCondition();
+                PathPatternsRequestCondition pathPatternsCondition = mappingInfo.getPathPatternsCondition();
```
Comment (lovesh-ap) [inline instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java:20]: Extract the two statements out of for loop

File: instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.Iterator;
+
+public class SpringHelper {
+    public static <T> void gatherURLMappings(T mapping){
+        try {
+            RequestMappingInfo mappingInfo = (RequestMappingInfo) mapping;
+            PatternsRequestCondition patternsCondition = mappingInfo.getPatternsCondition();
+            PathPatternsRequestCondition pathPatternsCondition = mappingInfo.getPathPatternsCondition();
+            for (RequestMethod method : mappingInfo.getMethodsCondition().getMethods()) {
+                if (patternsCondition != null)
```
Comment (lovesh-ap) [inline instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java:21]: missing curly braces, proper formatting.

File: instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                    for (String url : patternsCondition.getPatterns()) {
+                        URLMappingsHelper.addApplicationURLMapping(new ApplicationURLMapping(method.name(), url));
+                    }
+                else if (pathPatternsCondition != null) {
+                    for (PathPattern url : pathPatternsCondition.getPatterns()) {
+                        if (url != null)
```
Comment (lovesh-ap) [inline instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java:27]: missing curly braces, proper formatting.

File: instrumentation-security/spring-webmvc-3.1.0/src/main/java/org/springframework/web/servlet/handler310/SpringHelper.java (hunk 1 above)
Comment (IshikaDawda) [reply instrumentation-security/spring-webmvc-3.1.0/src/main/java/org/springframework/web/servlet/handler310/SpringHelper.java]: Fixed this in commit [c548fde](https://github.com/newrelic/csec-java-agent/pull/53/commits/c548fde01b1e9223a5d86171749fedcf35f7a406)

File: instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java (hunk 2 above)
Comment (IshikaDawda) [reply instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java]: Fixed this in commit [a89078f](https://github.com/newrelic/csec-java-agent/pull/53/commits/a89078fac42fdd90722642f69c2063d23227024c)

File: instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java (hunk 3 above)
Comment (IshikaDawda) [reply instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java]: Fixed this in commit [2875fd0](https://github.com/newrelic/csec-java-agent/pull/53/commits/2875fd03e64b3ffa9a2115244c19655c0c82bf0b).

File: instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java (hunk 4 above)
Comment (IshikaDawda) [reply instrumentation-security/spring-webmvc-5.3.0/src/main/java/org/springframework/web/servlet/handler530/SpringHelper.java]: Fixed this in commit [2875fd0](https://github.com/newrelic/csec-java-agent/pull/53/commits/2875fd03e64b3ffa9a2115244c19655c0c82bf0b).

**Pull request merged**